CMakeLists.txt

```diff
@@ -242,6 +242,9 @@ if(ENABLE_UNSUPPORTED_EXPERIMENTAL_POSSIBLY_BROKEN_WAYLAND)
             message(WARNING "On Wayland EGL is required for GL, disabling GL support")
             set(ENABLE_GL OFF CACHE BOOL "" FORCE)
         endif()
+        if(ENABLE_XLIB)
+            message(FATAL_ERROR "Xlib and Wayland support are mutually exclusive")
+        endif()
```

Comment:
I don't see why this is, both options should be able to co-exist depending on which windowing API the application uses.

Reply:
Because in `eglplatform.h` it's not possible to define both the Wayland typedefs and the XLib typedefs at the same time.